pkg/cgroup/v1.go

```diff
@@ -0,0 +1,200 @@
+package cgroup
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	permissions = 0555
+	MemLimit    = "memory.limit_in_bytes"
+	BlkioWeight = "blkio.weight"
+	CpuShares   = "cpu.shares"
+)
+
+type Subsystem struct {
+	name string
+	// params made to be slice just for the sake of possible future extension,
+	// when we might need more than one parameter from one group
+	params []string
+}
+
+type V1Service struct {
+	mu         sync.Mutex
+	subsystems []*Subsystem
+
+	root      string
+	rootGroup string
+	procsFile string
+}
+
+// NewV1Service sets up cgroup service to work with
+// hardcoded (on purpose) cpu, memory and blkio parameters
+func NewV1Service() *V1Service {
+	subsystems := map[string][]string{
+		"cpu":    {CpuShares},
+		"memory": {MemLimit},
+		"blkio":  {BlkioWeight},
+	}
+
+	s := &V1Service{
+		root:      "/sys/fs/cgroup",
+		rootGroup: "teleworker",
+		procsFile: "cgroup.procs",
+	}
+	for name, params := range subsystems {
+		sys := &Subsystem{
+			name:   name,
+			params: params,
+		}
+		s.subsystems = append(s.subsystems, sys)
+	}
+
+	return s
+}
+
+// NewV1Runner initialize cgroups service, but also runs
+// checks, creates root directory (if not presented)
+func NewV1Runner() (*V1Service, error) {
```

Comment:
Is this used anywhere? I can't seem to find whether the limits are actually applied to the job.

Reply:
With `InternalCallHandle` integrated into the lib, it'll make a lot more sense to handle cgroups completely from the main `teleworker` package too. Updated the `Job.Stop` method to remove the group directory.

`NewV1Runner`  is yet to be used from the server side, as it will check that all the subsystems are available and run cleanup goroutine (which is not yet implemented and I doubt it will be, but the idea still remains)